src/Microsoft.AspNet.Mvc.Formatters.Json/Internal/JsonArrayPool.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNet.Mvc.Formatters.Json.Internal
+{
+    public class JsonArrayPool<T> : IArrayPool<T>
```

Comment:
The `.Internal` namespace is that for us... Hidden from most users, and subject to change/disappearance in future releases.

We don't generally make things `internal` without a good reason.